lib/PlaidClient.js

```diff
@@ -210,6 +211,44 @@ Client.prototype.getTransactions =
     }, options, cb);
   };
 
+// getAllTransactions(String, Date, Date, Object?, Function)
+Client.prototype.getAllTransactions =
+  function(access_token, start_date, end_date, cb) {
+    var transactions = [];
+    var offset = 0, total_transactions = null;
+    var self = this;
+
+    var handleResponse = function(transactionsResponse) {
+      total_transactions = transactionsResponse.total_transactions;
+      transactions.push(...transactionsResponse.transactions);
+      offset += transactionsResponse.transactions.length;
+    };
+
+    Promise.coroutine(function*() {
+      while (true) {
+        if (transactions.length === total_transactions) {
+          break;
+        }
+
+        yield self.getTransactions(
```

Comment:
since we are in a generator, we can do something like

```node
const intermediaryTransactionsResponse  = yield self.getTransactions( /* ... */ );
handleResponse(intermediaryTransactionsResponse);
```

In that world, its probably best to have `handleTransactions` be inline within the loop.